runtime/testy/stateManager.py

```diff
@@ -129,10 +131,12 @@ def setValue(self, value, keys):
       currTime = time.time();
       for item in path:
         item[1] = currTime
-      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(value)
+      if send:
+        self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(value)
     except:
       error = StudentAPIKeyError(self.dictErrorMessage(i, keys, currDict))
-      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
+      if send:
+        self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
```

Comment:
Student code blocks until it receives a return value. 
That way students don't need to treat their main loop asynchronously